src/main.cpp

```diff
@@ -8,35 +8,80 @@
 
 #include <iostream>
 
+#include "argparser.h"
+
+
 
 using namespace std;
 
 int main(int argc, char *argv[]) {
+    argparser::ArgumentParser parser("yapgt (yet another photogrammetry tool)");
+
+    /* Calibration Arguments */
+    argparser::Argument a_loadCalibration("loadcalibration", "Filepath to load calibration from");
+    argparser::Argument a_saveCalibration("savecalibration", "Filepath to save calibration to");
+    argparser::Argument a_calibrationImages("calibrationImages", "Images for Calibration");
+    argparser::Argument a_calibrationCalibrateRow("calibrateRow", "corners row for Calibration");
+    argparser::Argument a_calibrationCalibrateColumn("calibrateColumn", "corners column for Calibration");
+
+    argparser::Argument a_matchImages("matchImages", "Images to use for matching");
+
+
+    parser.addArgument(&a_loadCalibration);
+    parser.addArgument(&a_saveCalibration);
+    parser.addArgument(&a_calibrationImages);
+    parser.addArgument(&a_calibrationCalibrateRow);
+    parser.addArgument(&a_calibrationCalibrateColumn);
+    parser.addArgument(&a_matchImages);
+
+    parser.parseArguments(argc, argv);
+
     Calibration calb = Calibration();
 
+    if (!(a_loadCalibration.isFound() || a_calibrationImages.isFound())) {
+        cout << "Neither " << a_loadCalibration.getName() << " nor " << a_loadCalibration.getName() << " was supplied" << endl;
+        return -1;
+    }
 
-    filesystem::path calibrationFile("./resources/calibration.xml");
-    filesystem::path path("./resources/chessboard_images");
+    if (a_loadCalibration.isFound()) {
+        calb.loadCalibration(filesystem::path(a_loadCalibration.getValue<string>()));
+    }
 
-    if (!filesystem::exists(calibrationFile)) {
+    if (a_calibrationImages.isFound()) {
```

Comment:
What happens, when loadCalibration and calibrationImages are specified? Right now the file would be loaded and then regenerated.

Reply:
This is correct. The question is, what behaviour do we except? An error message?

Reply:
An error message and exiting the program are enough.